macOS Release Build Failure

Happens when building #1326 in either Release or RelWithDebInfo using Xcode generator:

```
In file included from /Users/adamj/Code/Plasma/build/Sources/Plasma/PubUtilLib/plPhysX/CMakeFiles/plPhysX.dir/Release/cmake_pch.hxx:4:
In file included from /Users/adamj/Code/Plasma/Sources/Plasma/PubUtilLib/plPhysX/Pch.h:48:
In file included from /Users/adamj/Code/Plasma/Sources/Plasma/PubUtilLib/plPhysX/plPhysXAPI.h:55:
In file included from /Users/adamj/Code/Plasma/build/vcpkg_installed/x64-macos-plasma/include/PxPhysicsAPI.h:45:
In file included from /Users/adamj/Code/Plasma/build/vcpkg_installed/x64-macos-plasma/include/foundation/Px.h:37:
In file included from /Users/adamj/Code/Plasma/build/vcpkg_installed/x64-macos-plasma/include/foundation/PxSimpleTypes.h:40:
/Users/adamj/Code/Plasma/build/vcpkg_installed/x64-macos-plasma/include/foundation/PxPreprocessor.h:454:2: error: Exactly one of NDEBUG and _DEBUG needs to be defined!
#error Exactly one of NDEBUG and _DEBUG needs to be defined!
```

--- a/cmake/FindPhysX.cmake
+++ b/cmake/FindPhysX.cmake
@@ -229,8 +229,7 @@ macro(_find_physx_library SUFFIX)
         # The PhysX headers require that either _DEBUG or NDEBUG is always
         # defined (but not both at once), otherwise they produce an error.
         target_compile_definitions(${TARGET} INTERFACE
-            $<$<CONFIG:Debug>:_DEBUG>
-            $<$<NOT:$<CONFIG:Debug>>:NDEBUG>
+            $<IF:$<CONFIG:Debug>,_DEBUG,NDEBUG>
         )
     endif()
 

--- a/cmake/FindPhysX.cmake
+++ b/cmake/FindPhysX.cmake
@@ -228,12 +228,9 @@ macro(_find_physx_library SUFFIX)
 
         # The PhysX headers require that either _DEBUG or NDEBUG is always
         # defined (but not both at once), otherwise they produce an error.
-        # For release builds, NDEBUG is always automatically defined
-        # (either by CMake and/or the compilers).
-        # For debug builds, _DEBUG is automatically defined by MSVC,
-        # but for other compilers we need to define it ourselves.
         target_compile_definitions(${TARGET} INTERFACE
-            $<$<AND:$<CONFIG:Debug>,$<NOT:$<CXX_COMPILER_ID:MSVC>>>:_DEBUG>
+            $<$<CONFIG:Debug>:_DEBUG>
+            $<$<NOT:$<CONFIG:Debug>>:NDEBUG>
         )
     endif()
 

